src/pages/VenuePage/VenuePage.tsx

```diff
@@ -72,6 +86,15 @@ export const VenuePage: React.FC = () => {
   const venue = useSelector(currentVenueSelector);
   const venueRequestStatus = useSelector(isCurrentVenueRequestedSelector);
 
+  usePreloadAssets(
+    [
+      venue?.mapBackgroundImageUrl,
+      ...(venue?.rooms ?? []).map((room) => room?.image_url),
+    ]
+      .filter(isDefined)
+      .map((url) => ({ url }))
+  );
```

Comment:
you should `useMemo` here as it created a new array every time and will trigger `useEffect` of the hook.
```suggestion
  const assetsToPreload = useMemo(
    () =>
      [
        venue?.mapBackgroundImageUrl,
        ...(venue?.rooms ?? []).map((room) => room?.image_url),
      ]
        .filter(isDefined)
        .map((url) => ({ url })),
    [venue]
  );

  usePreloadAssets(assetsToPreload);
```

Reply:
I haven't looked at the definition of `usePreloadAssets`, but why do we `.map((url) => ({ url })),` map to an object with `url` rather than just a `string[]`?

Reply:
`usePreloadAssets()` allows for preloading different kinds of assets, just defaults to `image`

```
export interface PreloadAsset {
  url: string;
  as?: string;
  type?: string;
}
```